### PR TITLE
controller: cleanup complete deployments only

### PR DESCRIPTION
**Pull request by 0xmichalis**
Fixes https://github.com/kubernetes/kubernetes/issues/46932

@kubernetes/sig-apps-pr-reviews

### CODE REVIEW

File: pkg/controller/deployment/sync.go [hunk 1]
```diff
@@ -59,6 +59,14 @@ func (dc *DeploymentController) sync(d *extensions.Deployment, rsList []*extensi
 		return err
 	}
 
+	// Clean up the deployment when it's paused but not when it's proportionally scaled
+	// in the middle of an on-going rollout.
+	if d.Spec.Paused {
+		if err := dc.cleanupDeployment(oldRSs, d); err != nil {
```
Comment (janetkuo) [inline pkg/controller/deployment/sync.go]: Skip cleanup if rollbackTo field isn't `nil`?

Comment (kargakis) [inline pkg/controller/deployment/sync.go]: SG, updated

File: pkg/controller/deployment/sync.go [hunk 2]
```diff
@@ -59,6 +59,14 @@ func (dc *DeploymentController) sync(d *extensions.Deployment, rsList []*extensi
 		return err
 	}
 
+	// Clean up the deployment when it's paused but not when it's proportionally scaled
+	// in the middle of an on-going rollout.
```
Comment (janetkuo) [inline pkg/controller/deployment/sync.go]: Maybe you'd want to update the comment and mention the `rollbackTo == nil` logic

**Review by janetkuo — COMMENTED** [review]
One comment, lgtm otherwise

File: pkg/controller/deployment/sync.go (hunk 2 above)
Comment (kargakis) [inline pkg/controller/deployment/sync.go]: ok

**Review by janetkuo — APPROVED** [review]

**Pull request merged**